File: codechef/POLYREL.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class POLYREL {
	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int cases = Integer.parseInt(br.readLine());
		for(int i=0; i<cases; i++) {
			int vertices = Integer.parseInt(br.readLine());
			int answer = vertices;
			for(int j=0; j<vertices; j++) 
				br.readLine();
			while(vertices >= 6) {
				vertices = (int)Math.floor(vertices/2);
				answer += vertices;
			}
			System.out.println(answer);
		}
	}
}

File: codechef/POLYREL/polyrel.kt
package codechef

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val In = BufferedReader(InputStreamReader(System.`in`))
    val cases = In.readLine()!!.toInt()


    (1..cases).forEach {
        val nodes = In.readLine()!!.toInt()
        tailrec fun child(nodes : Int, children : Int) : Int {
            if(nodes <= 6)
                return children
            else {
                val possible = Math.floorDiv(nodes, 2)
                return child(Math.floorDiv(nodes, 2), children + Math.floorDiv(nodes, 2))
            }
        }
        (1..nodes).forEach { In.readLine() }
        println(child(nodes, nodes))
    }
}
